src/matching-service/index.js

```diff
@@ -1,171 +1,125 @@
+  
+  await channel.assertQueue(EASY_QUEUE, { durable: false, messageTtl: 30000 });
+  await channel.assertQueue(MEDIUM_QUEUE, { durable: false, messageTtl: 30000 });
+  await channel.assertQueue(HARD_QUEUE, { durable: false, messageTtl: 30000 });
+  
+  io.on("connection", (socket) => {
+    console.log(`Connection opened: ${socket.id}`);
+    const timeoutId = setTimeout(() => {
+      socket.disconnect(true);
+    }, TIMEOUT_MS);
+  
+    socket.on("match-request", async (message) => {
+
+      console.log("Received match request");
+
+      const parsedMessage = JSON.parse(message);
+      const { uid, complexity } = parsedMessage;
+      if (!uid || !complexity) {
+        io.to(socket.id).emit("match-failure", "Missing arguments");
+        return;
+      }
+  
+      const queueName = getQueue(complexity);
+  
+      if (!queueName) {
+        io.to(socket.id).emit("match-failure", "Invalid arguments");
+        return;
+      }
+  
+      const messageMatchRequest = {
+        isCancelRequest: false,
+        userId: uid,
+        socketId: socket.id,
+      }
+
+      console.log("send request message to queue");
+
+      channel.sendToQueue(queueName, Buffer.from(JSON.stringify(messageMatchRequest)));
+    });
+  
+    socket.on("cancel-request", async (message) => {
+      const parsedMessage = JSON.parse(message);
+      const { uid, complexity } = parsedMessage;
+      if (!uid || !complexity) {
+        // TODO: Emit different event for cancel failure (+ handle event in FE)
+        io.to(socket.id).emit("match-failure", "Missing arguments");
+        return;
+      }
+  
+      const queueName = getQueue(complexity);
+  
+      if (!queueName) {
+        io.to(socket.id).emit("match-failure", "Invalid arguments");
+        return;
+      }
+  
+      const messageCancelRequest = {
+        isCancelRequest: true,
+        userId: uid,
+        socketId: socket.id,
+      }
+  
+      channel.sendToQueue(queueName, Buffer.from(JSON.stringify(messageCancelRequest)));
+    });
+  
+    socket.on("disconnect", () => {
+      clearTimeout(timeoutId);
+      console.log("User has left");
+    });
   });
 
-  socket.on("cancel-request", async (message) => {
-    const parsedMessage = JSON.parse(message);
-    const { uid, complexity } = parsedMessage;
-    if (!uid || !complexity) {
-      console.log("Invalid request");
-      console.log(`uid: ${uid}`);
-      console.log(`complexity: ${complexity}`);
-      // TODO: Emit different event for cancel failure (+ handle event in FE)
-      io.to(socket.id).emit("match-failure", "Missing arguments");
-      return;
-    }
-
-    console.log(`cancel request: (${uid}, ${complexity})`);
-
-    const queueName = getQueue(complexity);
-    if (!queueName) {
-      io.to(socket.id).emit("match-failure", "Invalid arguments");
-      return;
-    }
-
-    // Removes the current message in the queue.
-    await channel.get(queueName);
+  listenToQueue(channel);
+  listenToReplies(channel, io);
```

Comment:
Perhaps you could rename these functions, or write some comments explaining what Queues and Replies they are listening to... I'm not really sure just by looking here